resources/authgear/templates/en/web/authflowv2/login.html

```diff
@@ -91,6 +102,9 @@ <h2 class="screen-description">
       class="flex flex-col gap-4 mt-8"
       method="post"
       novalidate
+      {{ if (or (and $show_phone_input $show_captcha_phone) (and $show_text_input (or $show_captcha_email $show_captcha_username)))}}
```

Comment:
The condition can be simplified into `if (or $show_captcha_phone $show_captcha_email $show_captcha_username)`.

Rationale: You have checked `$show_phone_input` or `$show_text_input` when you set `$show_captcha_*` to true. So `(and $show_phone_input $show_captcha_*` is redundant.